vllm/model_executor/models/phi3.py

```diff
@@ -0,0 +1,159 @@
+# coding=utf-8
+# Adapted from llama.py
+"""Inference-only Phi3 model code inherit from Llama.py"""
+
+from typing import Optional
+
+import torch
+from transformers import Phi3Config
+
+from vllm.attention import AttentionMetadata
+from vllm.config import CacheConfig, LoRAConfig
+from vllm.distributed import get_pp_group
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.models.llama import (LlamaAttention,
+                                              LlamaDecoderLayer,
+                                              LlamaForCausalLM, LlamaModel)
+
+from .utils import make_layers
+
+
+class Phi3Attention(LlamaAttention):
+
+    def __init__(
+        self,
+        config: Phi3Config,
+        quant_config: Optional[QuantizationConfig] = None,
+        bias: bool = False,
+        cache_config: Optional[CacheConfig] = None,
```

Comment:
Correct me if I'm wrong, but looks like the only difference from `LlamaAttention` is the `rope_scaling` attribute? I think it would be simpler to just edit `LlamaAttention` to set this attribute and pass `rope_scaling` from `Phi3DecoderLayer` then.

Reply:
Thanks @DarkLight1337 ! I’ve simplified it a bit. The main change is the `packed_modules_mapping`. The rope scaling changes will be covered in the following PR anyway https://github.com/vllm-project/vllm/pull/8254